packages/gatsby/src/internal-plugins/prod-404-500/gatsby-node.js

```diff
@@ -15,21 +17,22 @@ emitter.on(`CREATE_PAGE`, action => {
 
     const originalPage = originalStatusPageByStatus[status]
 
-    if (!originalPage) {
-      const storedPage = {
-        path: action.payload.path,
-        component: action.payload.component,
-        context: action.payload.context,
-        status,
-      }
+    const pageToStore = {
+      path: action.payload.path,
+      component: action.payload.component,
+      context: action.payload.context,
+      slices: action.payload.slices,
+      status,
+    }
 
-      originalStatusPageByStatus[status] = storedPage
-      originalStatusPageByPath[action.payload.path] = storedPage
+    if (!originalPage || !isEqual(originalPage, pageToStore)) {
```

Comment:
`isEqual` check here was added (on top of using `action.payload.slices`) so that cases like `src/pages/404.js` + `onCreatePage` hook that adds `slices` (or `context`) can work as otherwise original page would be used and we would ignore potential future edits to the page.